content/src/service/reporters/DeploymentReporterFactory.ts

```diff
@@ -0,0 +1,29 @@
+import { Environment, EnvironmentConfig } from "../../Environment"
+import { DeploymentReporter } from "./DeploymentReporter"
+import { SegmentIoAnalytics } from "./SegmentIoAnalytics"
+import { NoOpDeploymentReporter } from "./NoOpDeploymentReporter"
+import { SQSDeploymentReporter } from "./SQSDeploymentReporter"
+import { CompositeDeploymentReporter } from "./CompositeDeploymentReporter"
+
+export class DeploymentReporterFactory {
+
+    static create(env: Environment): DeploymentReporter {
+        let reporters: DeploymentReporter[] = []
+
+        if (env.getConfig(EnvironmentConfig.SQS_QUEUE_URL_REPORTING)) {
```

Comment:
Maybe we should check that all variables are set? Not just `SQS_QUEUE_URL_REPORTING`

Reply:
Or throw an error/warning if this variable is set and the others aren't